src/main/java/edu/cornell/kfs/fp/batch/service/impl/AwsAccountingXmlDocumentAccountingLineServiceImpl.java

```diff
@@ -0,0 +1,329 @@
+package edu.cornell.kfs.fp.batch.service.impl;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.coa.businessobject.Chart;
+import org.kuali.kfs.coa.businessobject.Account;
+import org.kuali.kfs.coa.businessobject.SubAccount;
+import org.kuali.kfs.coa.businessobject.ObjectCode;
+import org.kuali.kfs.coa.businessobject.SubObjectCode;
+import org.kuali.kfs.coa.businessobject.ProjectCode;
+import org.kuali.kfs.coa.service.ChartService;
+import org.kuali.kfs.coa.service.AccountService;
+import org.kuali.kfs.coa.service.SubAccountService;
+import org.kuali.kfs.coa.service.ObjectCodeService;
+import org.kuali.kfs.coa.service.SubObjectCodeService;
+import org.kuali.kfs.coa.service.ProjectCodeService;
+import org.kuali.kfs.coreservice.framework.parameter.ParameterService;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.sys.KFSConstants;
+
+import edu.cornell.kfs.fp.batch.xml.AccountingXmlDocumentAccountingLine;
+import edu.cornell.kfs.fp.batch.xml.DefaultKfsAccountForAws;
+import edu.cornell.kfs.fp.batch.xml.cloudcheckr.GroupLevel;
+import edu.cornell.kfs.fp.batch.service.AwsAccountingXmlDocumentAccountingLineService;
+import edu.cornell.kfs.fp.CuFPConstants;
+
+public class AwsAccountingXmlDocumentAccountingLineServiceImpl implements AwsAccountingXmlDocumentAccountingLineService {
+
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(AwsAccountingXmlDocumentAccountingLineServiceImpl.class);
+
+    protected ParameterService parameterService;
+    protected ChartService chartService;
+    protected AccountService accountService;
+    protected SubAccountService subAccountService;
+    protected ObjectCodeService objectCodeService;
+    protected SubObjectCodeService subObjectCodeService;
+    protected ProjectCodeService projectCodeService;
+
+    @Override
+    public AccountingXmlDocumentAccountingLine createAccountingXmlDocumentAccountingLine(GroupLevel costCenterGroupLevel,
+                                                                                         DefaultKfsAccountForAws defaultKfsAccountForAws) {
+        LOG.info("createAccountingXmlDocumentAccountingLine for " + costCenterGroupLevel.getGroupName());
+
+        String costCenterGroupValue = costCenterGroupLevel.getGroupValue();
```

Comment:
the GroupLevel object has a function isCostCenterGroupLevel(), I think one of the first should do is to call that function, and if it is false, throw an illegalArgumentException